modules/couchbase/src/main/java/org/testcontainers/couchbase/CouchbaseContainer.java

```diff
@@ -105,37 +108,79 @@
 
     private String urlBase;
 
+    private GenericContainer proxy;
+
     public CouchbaseContainer() {
-        super("couchbase/server:" + VERSION);
+        this("couchbase/server:" + VERSION);
     }
 
     public CouchbaseContainer(String containerName) {
         super(containerName);
+
+        withNetwork(Network.SHARED);
+        withNetworkAliases("couchbase-" + Base58.randomString(6));
+        setWaitStrategy(new HttpWaitStrategy().forPath("/ui/index.html"));
     }
 
     @Override
-    protected Integer getLivenessCheckPort() {
-        return getMappedPort(8091);
+    public Set<Integer> getLivenessCheckPortNumbers() {
+        return Sets.newHashSet(getMappedPort(8091));
     }
 
     @Override
-    public Set<Integer> getLivenessCheckPortNumbers() {
-        return Sets.newHashSet(getLivenessCheckPort());
+    @SneakyThrows
+    protected void doStart() {
```

Comment:
the trick here is:
1) expose all ports on Socat container before the start
1) start Socat container
1) for each additional port, start a sub-process with another socat, with mapping `internal port -> external port`
1) modify Couchbase's configuration, make it run on the same port numbers as the external ones but internally